### PR TITLE
TokenFilterStem: add support for searching by the original token conditionally

### PR DESCRIPTION
**Pull request by kou**
Users can search by the original token (not stemmed token) conditionally with this.

It's useful when users want to search by a stemmed word generally but users sometimes want to search by a exact (not stemmed) word. For example:

* Searching by a stemmed word returns many results.
* Stemming returns wrong result.
* Users want to find only records that have a exact (not stemmed) word.

### CODE REVIEW

**Comment by kou** [comment]
Cc: @HashidaTKS

**Review by komainu8 — APPROVED** [review]
+1

File: test/command/suite/token_filters/stem/keep_original/add.test [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#@on-error omit
+plugin_register token_filters/stem
+#@on-error default
+
+table_create Memos TABLE_NO_KEY
+column_create Memos content COLUMN_SCALAR ShortText
+
+table_create Terms TABLE_PAT_KEY ShortText \
+  --default_tokenizer TokenBigram \
```
Comment (HashidaTKS) [inline test/command/suite/token_filters/stem/keep_original/add.test:9]: I wondered why the stemming works when the tokenizer is Bigram, but I noticed that `Tokenbigram` uses white-space-separate for ascii when a normalizer is used. 
 (https://groonga.org/docs/reference/tokenizers/token_bigram.html)
This is the reason why it works, right?

**Review by HashidaTKS — APPROVED** [review]
LGTM

File: test/command/suite/token_filters/stem/keep_original/add.test (hunk 1 above)
Comment (kou) [reply test/command/suite/token_filters/stem/keep_original/add.test:9]: Right.

**Pull request merged**